packages/ethereum/contracts/HoprNetworkRegistry.sol

```diff
@@ -70,51 +87,82 @@ contract HoprNetworkRegistry is Ownable {
+   * hopr node peer id should always start with '16Uiu2HA' (0x3136556975324841) and be of length 53
    */
-  function selfRegister(string calldata hoprAddress) external mustBeEnabled returns (bool) {
+  function selfRegister(string calldata hoprPeerId) external mustBeEnabled returns (bool) {
+    require(
+      bytes(hoprPeerId).length == 53 && bytes32(bytes(hoprPeerId)[0:8]) == '16Uiu2HA',
+      'HoprNetworkRegistry: HOPR node peer id must be valid'
+    );
+    // get account associated with the given hopr node peer id, if any
+    address registeredAccount = nodePeerIdToAccount[hoprPeerId];
+    // if the hopr node peer id was linked to a different account, revert.
+    // To change a nodes' linked account, it must be deregistered by the previously linked account
+    // first before registering by the new account, to prevent hostile takeover of others' node peer id
+    require(
+      registeredAccount == msg.sender || registeredAccount == address(0),
+      'HoprNetworkRegistry: Cannot link a registered node to a different account'
+    );
+
+    // get multi address associated with the caller, if any
+    bytes memory registeredNodeMultiaddrInBytes = bytes(accountToNodePeerId[msg.sender]);
+    require(
+      registeredNodeMultiaddrInBytes.length == 0 ||
+        keccak256(registeredNodeMultiaddrInBytes) == keccak256(bytes(hoprPeerId)),
+      'HoprNetworkRegistry: Cannot link an account to a different node. Please remove the registered node'
+    );
+
     if (requirementImplementation.isRequirementFulfilled(msg.sender)) {
-      accountToNodeMultiaddr[msg.sender] = hoprAddress;
-      nodeMultiaddrToAccount[hoprAddress] = msg.sender;
-      emit Registered(msg.sender, hoprAddress);
+      // only update the list when no record previously exists
+      if (registeredNodeMultiaddrInBytes.length == 0) {
+        accountToNodePeerId[msg.sender] = hoprPeerId;
+        nodePeerIdToAccount[hoprPeerId] = msg.sender;
+        emit Registered(msg.sender, hoprPeerId);
+      }
       emit EligibilityUpdated(msg.sender, true);
       return true;
     }
 
-    if (bytes(accountToNodeMultiaddr[msg.sender]).length != 0) {
-      // if the account has a registration entry, keep the entry but update the eligibility
-      emit EligibilityUpdated(msg.sender, false);
-    }
+    emit EligibilityUpdated(msg.sender, false);
     return false;
   }
 
+  /**
+   * @dev Allows when there's already a multi address associated with the caller account, remove the link by deregistering
+   */
+  function selfDeregister() external mustBeEnabled returns (bool) {
+    string memory registeredNodeMultiaddr = accountToNodePeerId[msg.sender];
+    require(bytes(registeredNodeMultiaddr).length > 0, 'HoprNetworkRegistry: Cannot delete an empty entry');
+    delete accountToNodePeerId[msg.sender];
+    delete nodePeerIdToAccount[registeredNodeMultiaddr];
+    emit Deregistered(msg.sender);
```

Comment:
Why don't emit the EligibiltyUpdated event?

Reply:
The SC handles "registration" and "eligibility" seperately. 
- "Registration" refers to the link between an Ethereum address and a hopr peer Id. This may change when an old node stops working the staker wants to link a new hopr node. 
- "Eligibility" refers to if an Ethereum account owns certain nft, has certain amount of stake in the staking contract. 

In this case, when an ethereum account `selfDeregister`, it only breaks the link between an account and a node. The account's eligibility may still remain the same.